### DISCUSSION TITLE
Vue vs React

### DISCUSSION THREAD

**Discussion by maxkario** [Launch-X-Latam/MisionFrontEnd#11] [General]
Opino que seria ideal aprender React, ya que es el framework que mas se esta usando. Me gustaria de verdad aprender Vue pero como dije anteriormente serie ideal aprender React. 
Pero eso lo dejamos para la encuesta.

**Comment by OscarNoe** [comment]
Que sea react!!

**Reply by OscarNoe** [reply]
@romarpla por si aun no lo decides 👀

**Comment by JosafatJimenezB** [comment]
Vue es bueno para empezar ya que es facil de entender aunque tambien react es mas usado hoy en día

apoyo react :sparkles:

**Reply by OATS74** [reply]
voto por react

**Comment by ProDeveloperRoy** [comment]
React & Angular

**Comment by FluffyRaccon** [comment]
Básicamente funcionan muy similar, sólo que con la diferencia que VUE suele ser más rápido, pero menos robusto, se suele usar en landing pages. Yo no conozco ambos bien, pero apoyo la idea.

**Comment by Rommelpalma123** [comment]
que sea react

**Comment by geraodst117** [comment]
Definitivamente react, aun mas por sus hooks y que es "sencillo" hacerlo adaptativo para desarrollar para multiples tamaños de pantalla

**Comment by PipeMantilla** [comment]
React x8

**Comment by juan-martinezu** [comment]
React por favor

**Comment by AnaKwi** [comment]
React porque está entre Vue y Angular!! Así que opto por el de enmedio que es React.

**Comment by SebastianSamboni** [comment]
Que sea React

**Comment by juanprxn1** [comment]
React!!!

**Comment by aiirvizionz** [comment]
React definitivamente :D

**Comment by fabiancgc12** [comment]
Preferiría Vue, es mucho mas rápido y considero que React al ser no-opinado tiende a llevar a malas practicas o mil maneras de hacer una sola cosa que en otro proyecto se hace otra manera (véase formularios o manejo de estado global), y el uso de los hooks tiende a ser mas molesto a las alternativas de Vue puede tener, por ejemplo el arreglo de dependencias y olvidarte de agregarle un valor. 

 Además Vue puede agregarse a paginas ya existentes sin ningún problema :D

**Reply by maxkario** [reply]
Por eso prefiero Vue pero React se usa más en un ambito global.

**Reply by fabiancgc12** [reply]
Es que es mas fácil y cómodo. Yo he usado react y aunque al principio puede parecer genial, mientras mas lo comparas con los beneficios de otros framework muchas veces termina pareciendo que el desarrollador pelea contra React  para que haga lo que quieres u optimizarlo en lugar de facilitarte las cosas, tal y como hacen Vue o Svelte. Las ventajas de React es su gran ecosistema pero como comente, 1000 maneras de hacer una sola cosa también puede ser un problema

**Reply by fabiancgc12** [reply]
Aunque viendo los comentarios parece que llevo las de perder jaja

**Reply by snowingt** [reply]
Yo he visto ambos y la cosa esta en , si react tiene mas búsqueda de empleo pero eso es mas competencia que vue .
Y aveces la especialidad apremia en este caso estoy indeciso pero como estoy viendo react de a poquito pues asi prefiero react xD

**Reply by snowingt** [reply]
> Es que es mas fácil y cómodo. Yo he usado react y aunque al principio puede parecer genial, mientras mas lo comparas con los beneficios de otros framework muchas veces termina pareciendo que el desarrollador pelea contra React  para que haga lo que quieres u optimizarlo en lugar de facilitarte las cosas, tal y como hacen Vue o Svelte. Las ventajas de React es su gran ecosistema pero como comente, 1000 maneras de hacer una sola cosa también puede ser un problema

Que tan avanzado estas ? Quieres algun compañero para ir a la par ?

**Reply by fabiancgc12** [reply]
Pues es el framework con el que estoy mas familiarizado aunque no sabría calificar lo avanzado que estoy pero si sirve de algo lo que suelo utilizar: context, trabajo con typescript que es esencial, RTK o recoil, react hook form, react router y de libreria de componentes e usado antD pero ahora uso mantine. Al usarlo tanto pues se de sus pro y sus contras, o al menos en mi experiencia y la que he llegado a leer por la comunidad.

Como seria lo de un compañero para ir a la par?.

**Reply by GusAlhazred** [reply]
Yo pienso igual. 
Estuve investigando y, si bien React esta bueno y me encanto lo poco que vi en persona, para aprender te sirve mas otros de los framework (VUE, Angular, etc). Por lo que interprete, un poco por lo mismo que Js no esta tan bueno para aprender a programar. Js, al no ser un lenguaje fuertemente tipado y darte tantas libertades, te da la posibilidad de caer en malas practicas y cuando tengas que hacer cosas con bases de datos o trabajar con otros lenguajes, te toca desaprender esas cosas que Js te permitia. Con React pasa algo similar.
Plus lo que hablan de la competencia
Pero no parece una opinion muy popular y, al final si hacemos las cosas bien, cualquier framework viene bien. Y tampoco es que estamos votando entre Vue o Cobol, asi que dale que va hahaha

**Reply by Necrowolf98** [reply]
Pienso igual que usted estimado.
La sintaxis de Vue.js es más sencilla y fácil de entender más cuando se empieza aprender desde cero.
Cuando comprendamos Vue.js sería bueno migrar a React.

**Reply by FaraN1ght** [reply]
Yo estoy finalizando el React con stack MERN del sensei Fernando Herrera y luego pasaré al de NodeJs, son cursos super baratos, actualizados y útiles. Lo que he aprendido hasta ahora es que antes de ir de un Framework a otro o ver cuál es más complicado primero deben dominar o saber las bases de Javascript, si no pues andarán como ciegos sin entender muchas cosas ya que son Frameworks basados en Javascript.

Si luego les molesta lo sucio que puede ser la estructura de React pues en el curso recomiendan Clean Architecture que básicamente es ordenar bien las carpetas de componentes simulando los patrones de diseño. 

Finalmente complementar con Clean Code, buenas prácticas y Typescript.

**Comment by PetuchoDeveloper** [comment]
Prefiero mil veces React, lo estuve usando un poco y me encantó y Vue no me acabo de gustar
Asi que mi voto es a React.js

**Comment by iguelAL** [comment]
React

**Comment by R0nn1n18204** [comment]
react estaria bien a mi parecer

**Comment by Csv3019** [comment]
React

**Comment by AlbinoJg** [comment]
React
x10000

**Comment by ghost** [comment]
React es más conocido. Que sea React.

**Comment by IvanDevJS** [comment]
React porfavor

**Comment by DaniEstefez27** [comment]
React por favor

**Comment by bmanriqueInnova** [comment]
apoyo react tiene mucha salida laboral

**Reply by AnaKwi** [reply]
definitivamente, es de lo que más piden jaja

**Comment by snowingt** [comment]
que sea react

**Comment by JG3R** [comment]
vue

**Comment by gersonjeanpierre** [comment]
React

**Comment by loo-kuhs** [comment]
He usado ambos, así que no tengo problema, aunque me gustaría profundizar un poco Vue. 🥸

**Comment by LaMazorca** [comment]
Siendo sincero, no he utilizado ninguno de los dos, pero si lo he visto, y me parece más interesante el react

**Comment by JoselinMiel** [comment]
No conozco ninguno de los pero me iría por Vue

**Comment by Maickol304** [comment]
Tengo por entendido que react tiene mayor salida laboral, por ende que sea react

**Comment by jpampa-dev** [comment]
No se diga mas, react

**Comment by LeandroPereira23** [comment]
Me parecería perfecto aprender React ya que es una meta que tengo desde hace mucho tiempo, porque es una la bibloteca de Javascript más utilizada para el desarrollo frontend en el mundo.

**Comment by RamaAlves** [comment]
Comparto! Me parece genial ir por React

**Comment by iguelAL** [comment]
Y que tal Angular?

**Reply by RamaAlves** [reply]
El Mision Comander no lo puso entre las opciones. (No conozco Angular como para opinar)

**Comment by alexInn0va** [comment]
otro voto para React! he visto más ofertas laborales para React, en la encuesta de StackOverflow NodeJS y ReactJS son los frameworks más utilizados, Vue no entra al top 5  :(
https://survey.stackoverflow.co/2022/#web-frameworks-and-technologies

![2022-09-26_20-48-13](https://user-images.githubusercontent.com/114436403/192412330-6d213d14-13f1-457b-b71b-48a2210ff07a.jpg)

**Comment by DanteAllccahuaman** [comment]
React

**Comment by kmn04MX** [comment]
React sería perfecto

**Comment by GonzaloZubiri** [comment]
Me gustaria que sea React.

**Comment by Kunturwari** [comment]
Personalmente creo que React seria mejor, tanto porque tiene buen alcance laboral como su potencia y comunidad. Si bien tiene sus pros y contras como cualquier otro framework actualmente creo que seria la mejor opción para aprender.

**Comment by NestorMyrie** [comment]
React!!! lo bueno de react es que su estructura se presta para react native en dado caso que queramos hacer apps móviles en un futuro 👾👾

**Comment by rodri-gh** [comment]
React

**Comment by rgplazas** [comment]
react

**Comment by AleEscalante** [comment]
React me parece sin duda alguna más demandado en el campo actual, así que por ello apoyo que podamos aprender a usar React.

**Comment by alejx99** [comment]
REACT!

**Comment by VanValentine** [comment]
React.js para front y node.js para back pero no se me adelanten que unos apenas estamos verdes en esto :)

**Comment by M31C41** [comment]
No es por ser ególatra pero me gusta más mi idea de que sea Angular :v

**Comment by Beather1205** [comment]
React!

**Comment by Danielrojas20024** [comment]
Es mejor implementar React!

**Comment by Lucasgithub21** [comment]
React!!!

**Comment by Ness-GM** [comment]
React!

**Comment by fp2023** [comment]
JQuery

jajajaja

Entiendo que Vue sería más facil de aprender que React, pero por un tema laboral, son más las ofertas para React.
Por lo que prefiero React y después individualmente podemos aprender Vue.

**Comment by AnaPatino18** [comment]
React!!

**Comment by JuanDLF** [comment]
react!!!!

**Comment by Ignacioskm** [comment]
Yo creo que por demanda laboral debería ser React, saludos a todos.

**Comment by terrykofmaxim002** [comment]
yo inicio desde cero pero para ir conversando, aprendiendo de ustedes pongo mi comentario y los sigo leyendo :D 
Saludos desde Toluca, México

**Reply by MammaniNelsonD** [reply]
jajja estamos iguales, también que elijan ellos y toque lo que toque hno. 👍

**Reply by terrykofmaxim002** [reply]
jaja, así es, de igual forma vamos a aprender de los mejores bro y vamos a salir desarrollando  :D

**Comment by MammaniNelsonD** [comment]
Sera lo que tenga que ser... jajj 👍

**Comment by RafaelGarEst** [comment]
react ofrece mas oportunidades asi que yo voto por react

**Comment by ReyesBF** [comment]
React sin dudar

**Comment by jhon-jader** [comment]
React

**Comment by davjopalandri** [comment]
Let´s React!

**Comment by Danfel009** [comment]
React!

**Comment by FabyRazo2** [comment]
React

**Comment by GusAlhazred** [comment]
![go-banana-ralph-wiggum](https://user-images.githubusercontent.com/101288514/192697702-9003f2ef-975f-4ab9-bc62-0ede2150c019.gif)
~~Quien dijo Angular?~~

**Comment by jvray00** [comment]
El que sea más utilizado o solicitado por la industria, el Mission Commander  o quienes ya tienen experiencia deberían tener la respuesta. Creo que la idea es que al concluir el programa estemos lo más cercano posible a ocupar un puesto en el campo laboral.

**Comment by RafaelMartinCallataFlores** [comment]
React

**Comment by yamilmmoresco** [comment]
React sería más interesante.

**Comment by Linamar126** [comment]
Vue es un poco más sencillo que react, es mejor iniciar con Vue y posteriormente pasar a react. Voto por Vue.

**Comment by adidiergg** [comment]
React

**Comment by AngeloParra** [comment]
React!

**Comment by MarioStr1fe** [comment]
REACT estaria bien

**Comment by Ketote95** [comment]
Yo creo que React sería perfecto, porque desde mi punto de vista, es lo que más se está solicitando hoy en día.
